dpe/src/dpe_instance.rs

```diff
@@ -264,9 +268,11 @@ impl DpeInstance {
                 return Err(DpeErrorCode::InternalError);
             }
 
-            // TODO: The root node isn't a real node with measurements and
-            // shouldn't be in the cert. But we don't support DeriveChild yet,
-            // so this is the only node we can create to test cert creation.
+            // Skip adding the root node if it doesn't contain any measurements.
+            if curr.parent_idx == Context::ROOT_INDEX && !self.root_has_measurement.get() {
+                break;
+            }
```

Comment:
I'm wondering if it is relevant to always include the root measurement in the key derivation. Here's why:

Tree 1:

```
root(0x00) -> DeriveChild(0x00) -> DeriveChild(0x01)
                 ^
                 |
            ExtendTCI(0xAA
```

Tree 2:
```
root(0x00) -> DeriveChild(0x01)
     ^
     |
ExtendTCI(0xAA)
```

These will have the same key derived and the same certificate, but they technically took different boot paths.

When we talked about this previously, I wasn't thinking about the fact that you could technically call ExtendTCI on the root node.

Reply:
Yeah that makes sense to me. But maybe we could still exclude the root measurement if the DPE does not support ExtendsTci?